packages/dnb-eufemia/src/extensions/forms/Iterate/__tests__/Array.test.tsx

```diff
@@ -0,0 +1,279 @@
+  describe('using single render prop', () => {
+    describe('with primitive elements', () => {
+      it('should call renderers with each element value', async () => {
+        const renderProp = jest.fn()
+        render(
+          <Iterate.Array value={['first', 'second', 'third']}>
+            {renderProp}
+          </Iterate.Array>,
+        )
+
+        expect(renderProp).toHaveBeenCalledTimes(3)
+        expect(renderProp).toHaveBeenNthCalledWith(1, 'first', 0)
+        expect(renderProp).toHaveBeenNthCalledWith(2, 'second', 1)
+        expect(renderProp).toHaveBeenNthCalledWith(3, 'third', 2)
+      })
+    })
+    describe('with object elements and multiple render props', () => {
+      it('should call renderers with each element value', async () => {
+        const renderProp1 = jest.fn()
+        const renderProp2 = jest.fn()
+        render(
+          <Iterate.Array
+            value={[
+              { mem: 'A' },
+              { mem: 'B' },
+              { mem: 'C' },
+              { mem: 'D', second: '2nd' },
+            ]}
+          >
+            <Field.String elementPath="/" />
+            {renderProp1}
+            {renderProp2}
+          </Iterate.Array>,
+        )
+
+        expect(renderProp1).toHaveBeenCalledTimes(4)
+        expect(renderProp1).toHaveBeenNthCalledWith(1, { mem: 'A' }, 0)
+        expect(renderProp1).toHaveBeenNthCalledWith(2, { mem: 'B' }, 1)
+        expect(renderProp1).toHaveBeenNthCalledWith(3, { mem: 'C' }, 2)
+        expect(renderProp1).toHaveBeenNthCalledWith(
+          4,
+          { mem: 'D', second: '2nd' },
+          3,
+        )
+
+        expect(renderProp2).toHaveBeenCalledTimes(4)
+        expect(renderProp2).toHaveBeenNthCalledWith(1, { mem: 'A' }, 0)
+        expect(renderProp2).toHaveBeenNthCalledWith(2, { mem: 'B' }, 1)
+        expect(renderProp2).toHaveBeenNthCalledWith(3, { mem: 'C' }, 2)
+        expect(renderProp2).toHaveBeenNthCalledWith(
+          4,
+          { mem: 'D', second: '2nd' },
+          3,
+        )
+      })
+    })
+  })
+
+  describe('in DataContext', () => {
+    describe('with primitive elements', () => {
+      describe('referenced with path', () => {
+        it('should distribute values and receive callbacks on both iterate and context', async () => {
+          const dataContextOnChange = jest.fn()
+          const iterateOnChange = jest.fn()
+          render(
+            <DataContext.Provider
+              data={{
+                someList: ['foo', 'bar'],
+                otherValue: 'lorem ipsu',
+              }}
+              onChange={dataContextOnChange}
+            >
+              <Iterate.Array path="/someList" onChange={iterateOnChange}>
+                <Field.String elementPath="/" />
+                <Field.String path="/otherValue" />
+              </Iterate.Array>
+            </DataContext.Provider>,
+          )
+
+          const fields = document.querySelectorAll('input')
+          expect(fields).toHaveLength(4)
```

Comment:
```suggestion
          expect(fields).toHaveLength(4)
          
```

I like to have things that belong together in "blocks" for readability reasons ;)